workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/ViewRegistry.kt

```diff
@@ -153,6 +161,12 @@ public fun <RenderingT : Any> ViewRegistry.buildView(
  * Creates a [View] to display [initialRendering], which can be updated via calls
  * to [View.showRendering].
  *
+ * The returned view will have a [WorkflowLifecycleOwner] set on it, the caller _must_ ensure the
```

Comment:
I'm concerned about the must-attach-once part of the contract. e.g. I've just learned that our app is littered with cases where a window shown and dismissed so quickly fast that its decorView is never attached.

Reply:
Ah, you said below the attach actually isn't required.

Reply:
As long as the container actually destroys its lifecycle. Another mitigating factor is that, I believe, in practice nothing will ask for a `ViewTreeLifecycleOwner`, let alone actually start observing it, until the view is attached (because the assumption is that in 99% of cases the view itself won't have its own owner, and the view tree must be traversed to find one).